src/alt.js

```diff
@@ -34,6 +34,13 @@ function uid(container, name) {
   return key
 }
 
+/* for now just detect whether it's instance of Immutable.js */
+function isImmutable(state) {
```

Comment:
Another good thing about ImmutableUtil is that we can wrap the store in a way where we can make this check more robust.